### PR TITLE
In drop menu, add ability mark to drop all non-favorite and non-worn items

### LINKED ISSUE #29041: In drop menu, ability mark to drop all non-favorite and non-worn items
**Is your feature request related to a problem? Please describe.**  
When coming back from a looting raid, I have a lot of stuff in my inventory and I want to drop everything except the items I want to keep on me.  This is a bit tricky right now.  I know there's a tab option to drop all of a specific category but I really want to quickly drop all items I don't want and know when I leave my base I'm not missing vital gear.  

**Describe the solution you'd like**  
Using the current ability to mark inventory items as favorite, and I want to add the ability to drop all non-favorite and non-worn items.  On the drop menu, you can press a key to mark all non-favorite and non-worn items as items to be dropped on the right hand menu called "items to drop" . You can edit the "items to drop" list as usual before dropping the items, and pressing enter will confirm the drop list as usual. So basically to drop all non-favorite and non-worn items it'd be d, *, enter with default keybindings.

**Additional context**  
If this sounds like a something worth having in the game, I can work on coding up the feature as its been bugging me while playing the game and I knows code.

### PR DESCRIPTION
**Pull request by KarateSnoopy**
#### Summary

SUMMARY: Interface "Adds new command in drop menu to mark all non-favorite and non-worn items"

#### Purpose of change

Fixes #29041

#### Describe the solution
In the drop menu, you can now press the `,` key to mark all non-favorite and non-worn items to be dropped to the right hand column called "Items to Drop" . You can edit the "Items to Drop" column as usual before dropping the items, and pressing enter will confirm the drop as usual. Pressing `,` key again on the drop menu before confirming will toggle marking of these items just like other marking methods.

This means after favoriting each of the item you want to keep on you, you can now quickly type `d ,  enter` with default keybindings to drop all non-favorite and non-worn items which makes dropping all your loot very quick and easy and really works great when combined with zone sorting.

#### Additional context

Short YouTube walkthrough to see feature in action:
https://www.youtube.com/watch?v=vGB6PgiNNWU

### CODE REVIEW

**Comment by cafeoh** [comment]
Just a heads up, this is gonna conflict with my PR #29129.

I didn't see yours before, my bad. In my change I use the key `*` to toggle a whole stack as favorite. It's worth noting that the inventory selector (and its inventory entries) only expose the top item. This causes problem for stacks, where only the top (essentially random) item matters. I explicitly unstack favorite and non favorite items, so you can favorite a number of items you want to keep on you, and still leave the others in the "hoarding" pile.

I think the best course of action (what I had planned to do as soon my two other PRs are merged) is to keep existing binds to select the whole inventory, and simply toggle more of those items with subsequent presses (Both when selecting by category, or when moving all items in advanced inventory menu). 

**No item -> Only non-favorite items -> All items -> No item selected.**

Having a key to "Drop all except worn and favorite" might still be interesting in the multidrop menu though, I think using `,` by default would put it in line with the advanced inventory, and avoid conflicts with invlets, which can still be bound to `*`, and with my ideas.

I also set up the keybinds in other menus (we might want to favorite/unfavorite items while dropping, or picking stuff up), and in the advanced inventory menu.

**Comment by KarateSnoopy** [comment]
Ok, I can change it to ```,```

**Comment by cafeoh** [comment]
You've changed the wrong bind, and the action you define, `TOGGLE_FAVORITE` is also conflicting (you should probably change its name too)

As I said, I kind of had another idea in mind for `,` in multidrop menu (to select cyclically), so it's just a heads up since I'm very likely to submit a PR overwriting your changes in the future, so it's up to you.

I don't think it's a great idea to reference player in `inventory_column` (not really the right scope) and define player specific functions directly in it while carrying that reference everywhere. `g->u` is already accessible globally and references the actual player (which is the only instance of `player` to use menus, unlike NPCs). In this instance you could probably let the code that handles selection live in input.action ifelse chain. 

I've added helper functions and accessors in the column and selector class (I think some in the first PR, some still staged locally), but of course they're not accessible yet...

**Comment by KarateSnoopy** [comment]
Oops, you're right - wrong binding.  What hotkey do you suggest I use?  Do you want me to use `,` or something else?

I'll switch it to `g->u`  I was just following the existing pattern in the inventory_selector which took in a player object so I passed that player object to the column where I needed it.

**Comment by KarateSnoopy** [comment]
@cafeoh Can you take a look again?  Thanks

**Comment by codemime** [comment]
> I switched it to g->u. I was just following the existing pattern in the inventory_selector which took in a player object so I passed that player object to the column where I needed it but g->u is good by me

Neither of them looks good in the column (with `g->u` being worse). This pollutes the class with specific, scarcely used knowledge it should be unaware of. `is_favorite()`, `is_worn()`, and `get_all_nonfavorite_and_nonworn()` are unwelcome there too: way to specific, none of the column's buisness. If you need a custom way of retrieval, you could instead introduce something like either:

```c++
std::vector<const inventory_entry *> inventory_column::get_all_entries() const	// Gets filtered externally.
```

or:

```c++
std::vector<const inventory_entry *> inventory_column::get_entries(const std::function<bool(const inventory_entry&)>&) const	// Gets filtered internally.
```
The later is a bit more reusable and could be more efficient (no need to retrieve all and then toss it away), but the former can be of some use too.

File: src/inventory_ui.cpp [hunk 1]
```diff
@@ -275,7 +275,7 @@ void inventory_selector_preset::append_cell( const
 }
 
 void inventory_selector_preset::append_cell( const
-        std::function<std::string( const inventory_entry & )> &func,
+        std::function<std::string( const inventory_entry & )> &getTextFunc,
```
Comment (codemime) [inline src/inventory_ui.cpp]: An unrelated change which violates the [code style guide](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CODE_STYLE.md).

File: src/inventory_ui.cpp [hunk 2]
```diff
@@ -1919,6 +1945,24 @@ std::list<std::pair<int, int>> inventory_drop_selector::execute()
...
+                    return elem->chosen_count > 0;
+                } );
+
+                if( clear ) {
+                    count = max_chosen_count;
+                }
+            }
+
+            for( const auto &elem : selected ) {
+                set_chosen_count( *elem, count );
```
Comment (codemime) [inline src/inventory_ui.cpp]: This entire chunk is copy-pasted (except for the very first line).

**Comment by KarateSnoopy** [comment]
Thanks @codemime I'll get it fixed up

File: src/inventory_ui.cpp (hunk 1 above)
Comment (KarateSnoopy) [reply src/inventory_ui.cpp]: Fixed - reverted

File: src/inventory_ui.cpp (hunk 2 above)
Comment (KarateSnoopy) [reply src/inventory_ui.cpp]: Fixed - moved to a helper function

**Comment by KarateSnoopy** [comment]
Ok, I changed the design to use filter functions.  @codemime can you look again?  Thanks

File: src/inventory_ui.cpp [hunk 3]
```diff
@@ -1985,6 +2015,15 @@ std::list<std::pair<int, int>> inventory_drop_selector::execute()
             }
             set_chosen_count( *input.entry, count );
             count = 0;
+        } else if( input.action == "DROP_NON_FAVORITE" ) {
+            const auto filter_to_nonfavorite_and_nonworn = []( const inventory_entry & entry ) {
+                return ( entry.is_item() &&
```
Comment (Night-Pryanik) [inline src/inventory_ui.cpp]: Exterior brackets are redundant.

Comment (KarateSnoopy) [reply src/inventory_ui.cpp]: Thanks.  Fixed
